ipv8/attestation/trustchain/payload.py

```diff
@@ -168,27 +183,31 @@ class HalfBlockPairPayload(Payload):
     Payload for message that ships two half blocks
     """
 
-    format_list = ['74s', 'I', '74s', 'I', '32s', '64s', 'varlenI'] * 2
+    format_list = ['74s', 'I', '74s', 'I', '32s', '64s', 'varlenI', 'varlenI', 'Q'] * 2
 
     def __init__(self, public_key1, sequence_number1, link_public_key1, link_sequence_number1, previous_hash1,
-                 signature1, transaction1, public_key2, sequence_number2, link_public_key2, link_sequence_number2,
-                 previous_hash2, signature2, transaction2):
+                 signature1, block_type1, transaction1, timestamp1, public_key2, sequence_number2, link_public_key2,
+                 link_sequence_number2, previous_hash2, signature2, block_type2, transaction2, timestamp2):
```

Comment:
That's a lot of unnamed arguments. Thank God we don't have to worry if we misplace one of them because Python will figure out the type conversions :see_no_evil:

Reply:
In Python, we have to rely on proper unit tests to make sure that we do not make mistakes when passing arguments.

Reply:
:crying_cat_face: